<center><h1> <font color="blue">Prédiction et classification du cancer du sein</font></h1></center>

### 1. Importation des bibliothèques

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report

### 2. Lecture des données

In [3]:
data = pd.read_csv ("breast-cancer.csv")
data.head ()

mean_radius  mean_texture  mean_perimeter  mean_area  mean_smoothness   
0        17.99         10.38          122.80     1001.0          0.11840  \
1        20.57         17.77          132.90     1326.0          0.08474   
2        19.69         21.25          130.00     1203.0          0.10960   
3        11.42         20.38           77.58      386.1          0.14250   
4        20.29         14.34          135.10     1297.0          0.10030   

   diagnosis  
0          0  
1          0  
2          0  
3          0  
4          0

### 3. Analyse exploratoire

In [6]:
print("Dimension des données:", data.shape)

Dimension des données: (569, 6)


In [10]:
print(f"Type des données :\n{data.dtypes}")

Type des données :
mean_radius        float64
mean_texture       float64
mean_perimeter     float64
mean_area          float64
mean_smoothness    float64
diagnosis            int64
dtype: object


In [13]:
print(f"Nombre de valeurs manquantes par colonnes : \n{(data.isna().sum())}")

Nombre de valeurs manquantes par colonnes : 
mean_radius        0
mean_texture       0
mean_perimeter     0
mean_area          0
mean_smoothness    0
diagnosis          0
dtype: int64


In [17]:
print("Statistique descriptive :\n")
data.describe()

Statistique descriptive :



mean_radius  mean_texture  mean_perimeter    mean_area   
count   569.000000    569.000000      569.000000   569.000000  \
mean     14.127292     19.289649       91.969033   654.889104   
std       3.524049      4.301036       24.298981   351.914129   
min       6.981000      9.710000       43.790000   143.500000   
25%      11.700000     16.170000       75.170000   420.300000   
50%      13.370000     18.840000       86.240000   551.100000   
75%      15.780000     21.800000      104.100000   782.700000   
max      28.110000     39.280000      188.500000  2501.000000   

       mean_smoothness   diagnosis  
count       569.000000  569.000000  
mean          0.096360    0.627417  
std           0.014064    0.483918  
min           0.052630    0.000000  
25%           0.086370    0.000000  
50%           0.095870    1.000000  
75%           0.105300    1.000000  
max           0.163400    1.000000

#### Récapitulatif :
#### 3.1 Variabe à prédire : _diagnosis_
#### 3.2 Dimensions:
    - Nombre de lignes : 569
    - Nombre de colonnes : 6
#### 3.3 Types de données
    - features (mean_radius, mean_texture,..., mean_smoothness) : float64
    - target (diagnosis) : int64
#### 3.4 Valeurs manquantes
    - Aucune : étant donné qu'il y a aucune valeur manquante, on ne fera pas un nettoyage de données.

### Distribution de la variable cible : diagnosis

In [18]:
print("Distribution de la variable cible :\n")
data["diagnosis"].value_counts()

Distribution de la variable cible :



diagnosis
1    357
0    212
Name: count, dtype: int64